snyk/snyk-docker-plugin
# Implement the Go PCLN Table Parser

#### What does this PR do?

This commit / PR  adds a PCLN table parser to read from a Go binary and get
all files that were used to build that binary.
It is a requirement to read the packages from stripped Go binaries.

See [MYC-145](https://snyksec.atlassian.net/browse/MYC-145).

The main parsing code has been copied from the Go source and adapted to
our needs.

It does not integrate with anything yet, that work is tracked in
[MYC-146](https://snyksec.atlassian.net/browse/MYC-146).


#### How should this be manually tested?

One thing that would be good is if the binary-generating script could be run to make sure it's actually reproducible.

#### What are the relevant tickets?

[MYC-145](https://snyksec.atlassian.net/browse/MYC-145).
